javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/ClassLoaderMatcher.java

```diff
@@ -92,6 +95,18 @@ private static boolean canSkipClassLoaderByName(ClassLoader loader) {
       }
     }
 
+    private static boolean canSkipClassLoaderByPackagePrefix(ClassLoader loader) {
```

Comment:
I would just add this into `canSkipClassLoaderByName`

Reply:
That function uses the FQCN

Reply:
Yes, but still: check if `loader.getClass().getName()` is in a set of names or starts with a given prefix. Still sounds like `canSkipClassLoaderByName` to me :) You make decision based on classloader class name, no?

Reply:
fixed